src/screens/FeedScreen/Header.js

```diff
@@ -227,20 +324,47 @@ const _renderProfileNormal = ({
           <GlobalButton
             onPress={navigateToProfile}
             style={[styles.containerFeedProfile, {paddingBottom: 5}]}>
-            <View style={[styles.containerFeedName, {alignItems: 'flex-end'}]}>
-              <Text style={styles.feedUsername}>
-                {username || StringConstant.feedDeletedUserName}
-              </Text>
+            <View>
+              <View
+                style={[
+                  {
+                    flexDirection: 'row',
+                    alignItems: 'center'
+                  }
+                ]}>
+                <View
+                  style={{
+                    flexDirection: 'row'
+                  }}>
+                  <Text>{username || StringConstant.feedDeletedUserName}</Text>
+                  {!isSelf && (
+                    <React.Fragment>
+                      <View style={styles.point} />
+                      <TouchableOpacity onPress={() => onPressFollUnFoll(isFollow)}>
+                        <Text style={isFollow ? styles.textFollowing : styles.textFollow}>
+                          {isFollow ? 'Following' : 'Follow'}
+                        </Text>
+                      </TouchableOpacity>
+                    </React.Fragment>
+                  )}
+                </View>
 
-              <GlobalButton
-                buttonStyle={{marginLeft: 'auto', paddingBottom: 0, alignSelf: 'center'}}
-                onPress={onHeaderOptionClicked}>
-                {hideThreeDot ? null : (
-                  <View style={{zIndex: 1000}}>
-                    <ElipsisIcon width={4} height={14} fill={COLORS.blackgrey} />
-                  </View>
-                )}
-              </GlobalButton>
+                <GlobalButton
+                  buttonStyle={{
+                    marginLeft: 'auto',
+                    paddingBottom: 0,
+                    alignSelf: 'center'
+                  }}
+                  onPress={() => {
+                    refSheet.current.open();
+                  }}>
+                  {hideThreeDot ? null : (
+                    <View style={{zIndex: 1000}}>
+                      <ElipsisIcon width={4} height={14} fill={COLORS.blackgrey} />
+                    </View>
+                  )}
+                </GlobalButton>
+              </View>
             </View>
             <View style={[styles.containerFeedText, {paddingBottom: 0}]}>
               <Text style={styles.feedDate}>{calculateTime(time)}</Text>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

> Note: This review was outside of the patch, so it was mapped to the patch with the greatest overlap. Original lines [183-367]

The follow/unfollow functionality is implemented directly in the UI component. This can make the component less readable and harder to maintain. Consider moving this logic into a separate function or even a custom hook. This will improve the separation of concerns in your code.



<!-- This is an auto-generated comment by OSS CodeRabbit -->